docs/glossary.rst

```diff
@@ -0,0 +1,99 @@
+
+        For example, if you need a city, a cuisine, and a price range to recommend a restaurant, you can create  a restaurant form to do that. You can describe any business logic inside a form. For example, if you want to ask for a particular neighbourhood if a user mentions a large city like Los Angeles, you can write that logic inside the form.
+
+    Happy / Unhappy Paths
+        If your assistant asks a user for some information and the user provides it, we call that a happy path. Unhappy paths are all the possible edge cases of a bot. For example, the user refusing to give some input, changing the topic of conversation, or correcting something they said earlier.
+
+    Intent
+        Something that a user is trying to convey or accomplish (e,g., greeting, specifying a location).
+
+    Interactive Learning
+        A mode of training the bot where the user provides feedback to the bot while talking to it.
+
+        This is a powerful way to write complicated stories by enabling users to explore what a bot can do and easily fix any mistakes it makes.
+
+    NLG
+        Natural Language Generation (NLG) is the process of generating natural language messages to send to a user.
+
+        Rasa uses a simple template-based approach for NLG. Data-driven approaches (such as neural NLG) can be implemented by creating a custom NLG component.
+
+    Rasa NLU
+        Natural Language Understanding (NLU) deals with parsing and understanding human language into a structured format.
+
+        Rasa NLU is the part of Rasa that performs intent classification and entity extraction.
+
+    Pipeline
+        A Rasa bot's NLU system is defined by a pipeline, which is a list of NLU components (see "Rasa NLU Component") in a particular order. A user input is processed by each component one by one before finally giving out the structured output.
+
+    Policy
+        Policies make decisions on how conversation flow should proceed. At every turn, the policy which predicts the next action with the highest confidence will be used.  A Core model can have multiple policies included, and the policy whose prediction has the highest confidence decides the next action to be taken.
+
+    Rasa Core
+        The dialogue engine that decides on what to do next in a conversation based on the context.
+
+    Rasa NLU Component
+        An element in the Rasa NLU pipeline (see "Pipeline").
+
+        Incoming messages are processed by a sequence of components called a pipeline. A component can perform tasks ranging from entity extraction to intent classification to pre-processing.
+
+    Slot
+        A key-value store that Rasa uses to track information over the course of a conversation.
+
+        For the different types of slots, see :ref:`here <slot-classes>`.
+
+    Story
+        A template of a conversation between a user and a bot annotated with the intent / entities of the users' messages as well as the sequence of actions to be performed by the bot
```

Comment:
I think the word template is awkward here. It is  a conversation between a user and a bot, represented as intents, entities, and actions.

Reply:
Yeah, it does seem awkward, but I think what @akelad was getting at was probably that the stories don't contain the actual text of a conversation, but just the "normalized" intents, utterances, etc.

Reply:
But yeah, it seems redundant since we've already mentioned that it's annotated with the intent/entities and actions.

Reply:
yes but 'template' is misleading IMO. It's a real conversation, just represented at a higher level of abstraction. That's obviously too technical to say but that's the general iidea